From cd66f77ab7869894725dcd40ab0eb2dda09bf833 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Sun, 05 Feb 2023 11:22:37 +0200
Subject: [PATCH] [SW-114487] gaudi2: boot_info: modify address

This patch modifies device macros for boot_info base
from current ARC0 ICCM to SPSRAM(offset 20KB).

This is a done to avoid DMI access to ARC ICCM.
But as SPSRAM content is cleared during several resets(e.g.
hard_reset, prstn) a persistent copy to be maintained in ARC1 ICCM
to preserve any update in boot_info.

Change-Id: I01fb9c31bd4ee4432f49b9bc39fc847f67c847de
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index 86513ab..f5a6fe2 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -73,8 +73,8 @@
 #define PPBOOT_TEXT_BASE	(FW_SRAM_BASE + PPBOOT_START_OFFSET)
 #define PREBOOT_TEXT_BASE	(SDRAM_BASE + PREBOOT_START_OFFSET)
 
-#define ICCM_BOOT_INFO_OFFSET   0x18000
-#define BOOT_INFO_BASE          (ARC0_ICCM_BASE + ICCM_BOOT_INFO_OFFSET)
+#define BOOT_INFO_OFFSET	0x5000
+#define BOOT_INFO_BASE		(SP_SRAM_BASE + BOOT_INFO_OFFSET)
 
 /* In order not to conflict with ARC BOOTROM's usage of DCCM,
  * Let PPBoot start from the offset of 0x4030, for using DCCM
diff --git a/arch/gaudi2/arm/common.h b/arch/gaudi2/arm/common.h
index dab3257..33f957e 100644
--- a/arch/gaudi2/arm/common.h
+++ b/arch/gaudi2/arm/common.h
@@ -60,9 +60,9 @@
 
 #define HL_INIT_SP_SRAM_RETRIES		50
 
-#define ICCM_BOOT_INFO_OFFSET		0x18000
+#define BOOT_INFO_OFFSET		0x5000
 #define ARC0_ICCM			0x3FF7000000
-#define BOOT_INFO_BASE			(ARC0_ICCM + ICCM_BOOT_INFO_OFFSET)
+#define BOOT_INFO_BASE			(HL_SP_SRAM_BASE + BOOT_INFO_OFFSET)
 
 /*
  * Set ARCs in halt state before reset
